src/journal-remote/journal-remote-parse.c

```diff
@@ -487,7 +487,7 @@ int process_source(RemoteSource *source, bool compress, bool seal) {
         target = source->size;
         while (target > 16 * LINE_CHUNK && remain < target / 2)
```

Comment:
I think the change should be here instead: `remain` should be replaced by `source->filled`. At this point we don't care about `remain` at all, only about `source->filled`.